includes/wikia/services/UserService.class.php

```diff
@@ -175,10 +175,23 @@ private function getUsersFromCacheByName( &$names ) {
 
 	private function getUserFromMemCacheById( $id ) {
 		$cacheIdKey = wfSharedMemcKey( "UserCache:".$id );
-		if ( ( $value = F::app()->wg->memc->get( $cacheIdKey ) ) !== false ) {
-			//cache locally
-			$this->cacheLocalUser( $value );
-			return $value;
+		$value = F::app()->wg->memc->get( $cacheIdKey );
+
+		if ( $value !== false ) {
+			if ( $value instanceof User ) {
+				//cache locally
+				$this->cacheLocalUser( $value );
+				return $value;
+			} else {
+				Wikia\Logger\WikiaLogger::instance()->debug(
```

Comment:
Could this case return false as well or it's not possible? This would simplify returning, I guess :ghost: